src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,32 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+	public final String value;
+    private boolean isPrivate;
+    
+    public Contact(String value, boolean isPrivate) {
+    	this.value = value;
+    	this.isPrivate = isPrivate;
+    }
+
+	@Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Address // instanceof handles nulls
```

Comment:
should compare with an intenceof Contact.
Can override in subclasses to their respective type
If you override in subclasses make sure to override the hashCode() too